lib/network.cpp

```diff
@@ -168,9 +168,13 @@ int resolve_hostname(const char* hostname, sockaddr_storage &ip_addr) {
     hints.ai_protocol = IPPROTO_TCP;
     int retval = getaddrinfo(hostname, NULL, &hints, &res);
     if (retval) {
-        char buf[256];
-        snprintf(buf, sizeof(buf), "%s: getaddrinfo", time_to_string(dtime()));
-        perror(buf);
+        char buf[512];
+        snprintf(buf, sizeof(buf), "%s: getaddrinfo(%s) returned %s", time_to_string(dtime()), hostname, gai_strerror(retval));
```

Comment:
This could print "(date): getaddrinfo(host) returned The requested socket type is not supported". That looks a bit unfamiilar. Can't you just stick to the classic layout of "${function}: ${error cause}" rather than "${function} returned ${error cause}".